packages/fxa-shared/test/db/models/auth/prune-token.spec.ts

```diff
@@ -0,0 +1,161 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { expect } from 'chai';
+import { StatsD } from 'hot-shots';
+import { Knex } from 'knex';
+import 'mocha';
+import 'reflect-metadata';
+import sinon from 'sinon';
+import { PruneTokens, SessionToken } from '../../../../db/models/auth';
+import { MetaData } from '../../../../db/models/auth/metadata';
+import { uuidTransformer } from '../../../../db/transformers';
+import { ILogger } from '../../../../log';
+import { defaultOpts, testDatabaseSetup } from '../../helpers';
+
+describe('PruneTokens', () => {
+  const sandbox = sinon.createSandbox();
+  const pruneInterval = 50;
+  const maxJitter = 101;
+  const statsd = sandbox.spy({
+    increment: () => {},
+  });
+  const log = sandbox.spy({
+    error: () => {},
+  });
+  let knex: Knex;
+  let pruneTokens: PruneTokens;
+
+  before(async () => {
+    knex = await testDatabaseSetup({
+      ...defaultOpts,
+      auth: true,
+      oauth: false,
+      profile: false,
+    });
+
+    await MetaData.query().insert({
+      name: 'sessionTokensPrunedUntil',
+      value: '10',
+    });
+
+    await MetaData.query().insert({
+      name: 'sessionTokensLastPrunedAt',
+      value: '10',
+    });
+  });
+
+  after(async () => {
+    await knex.destroy();
+  });
+
+  beforeEach(async () => {
+    pruneTokens = new PruneTokens(
+      pruneInterval,
+      statsd as unknown as StatsD,
+      log as unknown as ILogger
+    );
+    await SessionToken.query().delete().execute();
+
+    // Add a dummy token that was created 10s ago
+    await insertToken(Date.now() - 1e4, Date.now() - 1e4);
```

Comment:
We could also insert a token that isn't expected to be pruned and ensure it is still there.

Reply:
Good call, I'll add that too. Admittedly, the tests are a bit light. For example I don't tests that codes are also getting cleared out properly... Maybe I should?